src/components/Card.js

```diff
@@ -1,58 +1,121 @@
-        deleteButton.addEventListener('click', () => this._handleDeleteCard());
+        this._likeButton.addEventListener('click', this._handleLikeClick);
 
         // Add event listener for image
         this._imageWindow = this._element.querySelector('.card__image');
         this._imageWindow.addEventListener('click', () => this._handleImageClick({link: this._link, name: this._name}));
     }
 
-    _handleLikeIcon = () => {
-        // Add active class to card's like button
-        this._likeButton.classList.toggle('card__like-button_active');
+    // Check if the card is liked by the user
+    isLiked() {
+        return this._cardLikes.some((cardLike) => cardLike._id === this._userId);
+    }
+
+    // Update the card's number of likes
+    setLikes(likes) {
+        this._cardLikes = likes;
+        this._renderLikes();
+    }
+
+    // Display the number of likes
+    _displayLikeCount() {
+        return this._likeCounter.textContent = this._cardLikes.length;
+    }
+
+    // Display the like icon based on its status
+    _renderLikes() {
+        if (this.isLiked()) {
+            this._likeButton.classList.add('card__like-button_active');
+        } else {
+            this._likeButton.classList.remove('card__like-button_active');
+        }
+        this._displayLikeCount();
+    }
+
+    // Activate the like icon
+    addLikeIcon() {
+        this._likeButton.classList.add('card__like-button_active');
+    }
+
+    // Deactivate the like icon
+    removeLikeIcon() {
+        this._likeButton.classList.remove('card__like-button_active');
     }
 
-    _handleDeleteCard() {
+    // Display bin icon on cards created by the user
+    _handleDeleteButton() {
+
+        // Check if the user id matches the card owner's id
+        if (this._userId !== this._cardOwnerId) {
+
+            // If it does, remove the delete button
+            this._deleteButton.remove();
+        }
+    }
+
+    // Handle the deletion of cards
+    deleteCard() {
+
         // Remove element from the DOM on click
         this._element.remove();
 
         // Remove the link to the DOM element
         this._element = null;
     }
 
+    // Get card template
     _getTemplate() {
-
-        // Get card template
+        
         return document
             .querySelector(this._cardSelector)
             .content.querySelector('.card')
             .cloneNode(true);
     }
 
+    // Render the card
     getView() {
         
         // Create card template
         this._element = this._getTemplate();
         
+        // Display the bin icon on the user's cards
+        this._deleteButton = this._element.querySelector('.card__delete-button');
+        this._handleDeleteButton();
+        this._deleteButton.addEventListener('click', () => this._handleDeleteClick());
+        
+        // Display the number of likes
+        this._likeCounter = this._element.querySelector('.card__like-number');
+        this._displayLikeCount();
```

Comment:
There is no need to call this method, since you do that in `_renderLikes` method